src/main/java/seedu/address/model/cheese/UniqueCheeseList.java

```diff
@@ -93,6 +96,20 @@ public void setCheeses(List<Cheese> cheeses) {
         internalList.setAll(cheeses);
     }
 
+    public Set<CheeseId> getUnassignedCheeses(CheeseType cheeseType, Quantity quantity) {
+        Set<CheeseId> cheeses = new HashSet<>();
+
+        for (int i = 0; i < internalList.size() && quantity != null; i++) {
+            Cheese cheese = internalList.get(i);
+            if (!cheese.getAssignStatus() && cheese.isSameType(cheeseType)) {
+                quantity = quantity.decreaseQuantity();
```

Comment:
The logic is good :) possibly can do 2 things
1. Sort by maturity date too (to ensure that the assigned cheeses are matured)
2. Can just use a counter to count the cheeses instead of using `decreaseQuantity()`, seems a bit sketchy haha
3. I think the `setCheese` should be done else where maybe in the command (using `model.setCheese()`). The `getUnassignedCheeses` should only get the cheeses without updating anything yet

Reply:
Don't think we should do a counter, as I was thinking of setting quantity's value to be private. 
3. setCheese , yes will be moving it out. realized that I might have not considered an edge case

Reply:
I'm not sure if it would be benefitial for us to set quantity's value to private (because it is already final so we can't rly change it anyways). But if you do have a purpose for it, then I'm alright with it

Reply:
I'm not sure if it would be benefitial for us to set quantity's value to private (because it is already final so we can't rly change it anyways). But if you do have a purpose for it, then I'm alright with it